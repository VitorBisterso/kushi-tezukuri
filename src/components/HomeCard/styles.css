.homeCard-container {
  position: relative;
  padding: 0;
  width: 15vw;
  height: 15vw;
  background-color: var(--COLOR-WHITE);
  border-radius: var(--BORDER_RADIUS-BIG);
  border: 1px var(--COLOR-SHADOW) solid;
  cursor: pointer;
  transition: box-shadow 0.2s linear;
  box-shadow:
    0px 2px 1px -1px rgb(0 0 0 / 20%),
    0px 1px 1px 0px rgb(0 0 0 / 14%),
    0px 1px 3px 0px rgb(0 0 0 / 12%);
}

.homeCard-container:hover {
  border: 2px var(--COLOR-SHADOW) medium;
  box-shadow: 5px 10px 15px var(--COLOR-GOLD);
  transition: 0.2s linear;
}

.homeCard-image {
  width: 100%;
  height: 100%;
  object-fit: cover;
  filter: saturate(1.5);
  border-radius: var(--BORDER_RADIUS-BIG);
}

.homeCard-title {
  position: absolute;
  top: 35%;
  left: 50%;
  transform: translate(-50%, -50%);
  z-index: 99;
  font-family: 'Mali-Bold';
  font-size: var(--TEXT-BIG);
  color: var(--COLOR-WHITE);
  background-color: var(--COLOR-SHADOW);
  text-align: center;
  width: 100%;
  word-break: break-word;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
}

@media (max-width: 767px) {
  .homeCard-container {
    width: 25vw;
    height: 25vw;
  }

  .homeCard-title {
    font-size: var(--TEXT-REGULAR);
  }
}
