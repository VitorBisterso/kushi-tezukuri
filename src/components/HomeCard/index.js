import React from 'react'
import PropTypes from 'prop-types'

import './styles.css'

const HomeCard = ({ image, title, onClick }) => (
  <button className="homeCard-container" onClick={onClick} type="button">
    <img className="homeCard-image" src={image} alt={`${title}-homeCard`} />
    <p className="homeCard-title">{title}</p>
  </button>
)

HomeCard.propTypes = {
  image: PropTypes.node.isRequired,
  title: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default HomeCard
