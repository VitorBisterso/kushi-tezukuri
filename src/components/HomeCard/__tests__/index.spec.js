import React from 'react'
import { shallow } from 'enzyme'

import HomeCard from '../index'

import image from '../../../assets/img/mask.jpg'

describe('HomeCard', () => {
  const clickFn = jest.fn()
  it('should render correctly', () => {
    const component = shallow(
      <HomeCard title="Test" image={image} onClick={clickFn} />
    )

    expect(component).toMatchSnapshot()
  })
  it('should call onClick function', () => {
    const component = shallow(
      <HomeCard title="Test" image={image} onClick={clickFn} />
    )

    component.find('button').simulate('click')
    expect(clickFn).toHaveBeenCalled()
  })
})
