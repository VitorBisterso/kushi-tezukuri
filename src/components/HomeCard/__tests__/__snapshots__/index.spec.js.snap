// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeCard should render correctly 1`] = `
<button
  className="homeCard-container"
  onClick={[MockFunction]}
  type="button"
>
  <img
    alt="Test-homeCard"
    className="homeCard-image"
    src=""
  />
  <p
    className="homeCard-title"
  >
    Test
  </p>
</button>
`;
